build/maven_test.go

```diff
@@ -72,3 +74,72 @@ func getExpectedMavenBuildInfo(t *testing.T, filePath string) entities.BuildInfo
 	assert.NoError(t, json.Unmarshal(data, &buildinfo))
 	return buildinfo
 }
+
+func TestExtractMavenPath(t *testing.T) {
+	service := NewBuildInfoService()
+	mavenBuild, err := service.GetOrCreateBuild("build-info-maven-test", "1")
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, mavenBuild.Clean())
+	}()
+	testdataDir, err := filepath.Abs(filepath.Join("testdata"))
+	assert.NoError(t, err)
+	// Create maven project
+	projectPath := filepath.Join(testdataDir, "maven", "project")
+	tmpProjectPath, cleanup := testdatautils.CreateTestProject(t, projectPath)
+	defer cleanup()
+	// Add maven project as module in build-info.
+	mavenModule, err := mavenBuild.AddMavenModule(tmpProjectPath)
+	assert.NoError(t, err)
+
+	allTests := []struct {
+		mavenVersionResultFirstLine  string
+		mavenVersionResultSecondLine string
+		mavenVersionResultThirdLine  string
+	}{
+		{"Maven home: /test/is/good", "Home: /test/is/not/good", "Mvn Home:= /test/is/not/good"},
+		{"Home: /test/is/not/good", "Maven home: /test/is/good", "Mvn Home:= /test/is/not/good"},
+		{"Mvn Home:= /test/is/not/good", "Home: /test/is/not/good", "Maven home: /test/is/good"},
+	}
+
+	for _, test := range allTests {
+		var mavenVersionFullResult []string
+		mavenVersionFullResult = append(mavenVersionFullResult, test.mavenVersionResultFirstLine, test.mavenVersionResultSecondLine, test.mavenVersionResultThirdLine)
+		data1 := bytes.Buffer{}
+		for _, i := range mavenVersionFullResult {
+			data1.WriteString(i)
+			data1.WriteString("\n")
+		}
+		mavenHome, err := mavenModule.extractMavenPath(data1)
+		assert.NoError(t, err)
+		if utils.IsWindows() {
+			assert.Equal(t, "\test\\is\\good", mavenHome)
+		} else {
+			assert.Equal(t, "/test/is/good", mavenHome)
+		}
+	}
+	assert.NoError(t, err)
+}
+
+func TestGetExecutableName(t *testing.T) {
+	service := NewBuildInfoService()
+	mavenBuild, err := service.GetOrCreateBuild("build-info-maven-test", "1")
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, mavenBuild.Clean())
+	}()
+	testdataDir, err := filepath.Abs(filepath.Join("testdata"))
+	assert.NoError(t, err)
+	// Create maven project
+	projectPath := filepath.Join(testdataDir, "maven", "project")
+	tmpProjectPath, cleanup := testdatautils.CreateTestProject(t, projectPath)
+	defer cleanup()
+	// Add maven project as module in build-info.
+	mavenModule, err := mavenBuild.AddMavenModule(tmpProjectPath)
+	assert.NoError(t, err)
+
+	mvnHome, err := mavenModule.getExecutableName()
+	assert.NoError(t, err)
+	assert.Equal(t, "mvn", mvnHome)
+
```

Comment:
This unit test can be simplified by focusing on the target function. There is no need to read a Maven project and add a Maven module. For example, the following is enough for testing GetExecutableName when wrapper==false:
```go
	mavenModule := MavenModule{extractorDetails: &extractorDetails{}}
	mvnHome, err := mavenModule.getExecutableName()
	assert.NoError(t, err)
	assert.Equal(t, "mvn", mvnHome)
```

When the code is simple, you can easily cover more flows such as useWrapper==true:
```go
	mavenModule = MavenModule{extractorDetails: &extractorDetails{useWrapper: true}}
```